src/lib.rs

```diff
@@ -33,6 +33,33 @@ use std::fmt;
 use std::io;
 use std::ops;
 
+macro_rules! convert_u8_to_u32 {
+    (ref mut $count:expr, $data:expr) => {
+        unsafe { std::mem::transmute::<&mut [u8; $count], &mut [u32; $count / 4]>($data) }
+    };
+}
```

Comment:
This looks not right. The alignment might be different between a `&[u8]` and `&[u32]`, so this is probably UB.

Reply:
Also endianess plays a role here. I don't think that this will work out.